client/java/src/main/java/com/linecorp/centraldogma/client/CentralDogma.java

```diff
@@ -535,8 +586,41 @@
      *
      * @param executor the {@link Executor} that executes the {@link Function}
      */
+    default <T, U> Watcher<U> fileWatcher(String projectName, String repositoryName,
+                                  Query<T> query, Function<? super T, ? extends U> function, Executor executor) {
+        return fileWatcher(projectName, repositoryName, query, function, executor, null);
+    }
+
+    /**
+     * Returns a {@link Watcher} which notifies its listeners after applying the specified
+     * {@link Function} when the result of the given {@link Query} becomes available or changes. e.g:
+     * <pre>{@code
+     * Watcher<MyType> watcher = client.fileWatcher(
+     *         "foo", "bar", Query.ofJson("/baz.json"),
+     *         content -> new ObjectMapper().treeToValue(content, MyType.class), executor,
+     *         WatchOptions.builder().timeoutMillis(60000).errorOnEntryNotFound(true).build());
+     *
+     * watcher.watch((revision, myValue) -> {
+     *     assert myValue instanceof MyType;
+     *     ...
+     * });}</pre>
+     *
+     * @param executor the {@link Executor} that executes the {@link Function}.
+     *        if {@code executor} is null, {@link Function} executed by a blocking task executor.
+     * @param watchOptions the {@link WatchOptions} that is options on waiting the matched files.
+     *        if {@code watchOptions} is null, use {@link WatchOptions#defaultOptions()}.
+     *        if {@link WatchOptions#errorOnEntryNotFound} is true and entry doesn't exist,
+     *        {@link Watcher#initialValueFuture()} will be completed with {@link EntryNotFoundException}.
+     *        {@link WatchOptions#timeoutMillis} is the waiting time about the matched files have a change.
+     *        If any no changes during {@link WatchOptions#timeoutMillis}, re-wait the matched files.
+     *        <p>Note: Legacy client only support {@link WatchOptions#timeoutMillis}</p>
+     */
+
     <T, U> Watcher<U> fileWatcher(String projectName, String repositoryName,
-                                  Query<T> query, Function<? super T, ? extends U> function, Executor executor);
+                                  Query<T> query,
+                                  Function<? super T, ? extends U> function,
+                                  @Nullable Executor executor,
```

Comment:
Conventionally, we usually don't take a nullable parameter in our public API. 
Could you revert the change, please? which means that we should revert this change:
https://github.com/line/centraldogma/pull/650/files#diff-f08760fa027ab0e795025c0410578002d9b1b5313714f83c79940f2d1b174807R569